src/chain/validity.rs

```diff
@@ -0,0 +1,56 @@
+use ethereum_types::*;
+use std::fmt::Display;
+
+#[allow(clippy::large_enum_variant)]
```

Comment:
But then I would have to unify them under one `ValidationError` for the purposes of execution function anyway.

Besides, it is not a hot path. In fact, we should almost never see this in practice - if a peer issues incorrect block it will be kicked out of P2P network really fast.

Reply:
👌